README.md

```diff
@@ -88,21 +93,66 @@ for development, specifically the table storage component which is not supported
 
    See [bug raised with the library](https://github.com/hey-red/Mime/issues/36) for more info.
 
+### Setting up the database and Identity Provider
+
+The service can be started against a set of non-existent database. If no pre-existing `content` or `statistics` 
+databases yet exist on the target SQL Server instance, starting up the service will create them and provide the basic
+starting data to get up and running.
+
+Similarly, the service can be started alongside a local Identity Provider called [Keycloak]() 
+
 ### Running the backend
 
 1. Add the following to your `hosts` file:
 
    ```
    127.0.0.1    db
    127.0.0.1    data-storage
-   ```
 
-2. If using SQLServer (instead of Docker), you should create your public frontend databases:
+2. Ensure you have Azure Storage Emulator running.
+
+3. Start the locally-provided Identity Provider or configure your own.
 
-   - `content`
-   - `statistics`
+Out of the box, the project supplies a Docker container Identity Provider called [Keycloak](https://www.keycloak.org/) 
+but any OpenID Connect compatible Identity Provider can be used e.g. Active Directory. It must have Implicit Flow enabled 
+and be using the OpenID Connect protocol. It must be set to issue ID Tokens.
 
-3. Ensure you have Azure Storage Emulator running.
+- To run the out-of-the-box Keycloak Identity Provider:
+
+  ```
+  cd useful-scripts
+  ./run.js idp
+  ```
+
+To then get the other service components to use Keycloak, set the following environment variable:
+```
+IdpProviderConfiguration=Keycloak
+```
+And additionally if wanting to set up a set of Keycloak users automatically in the service in order to start using 
+the service against an empty set of databases, set the following environment variable:
+```
+BootstrapUsersConfiguration=KeycloakBootstrapUsers
+```
+The effect of setting these 2 environment variables together will allow authentication of users with Keycloak and those
+users specified within the 
+`src\GovUk.Education.ExploreEducationStatistics.Admin\appsettings.KeycloakBootstrapUsers.json` will automatically be 
+able to be users to access the system as "BAU Users" - effectively able to bootstrap the rest of the data needed to use 
```

Comment:
I think "bootstrap the rest of the data needed to use the service" is a bit confusing here because it made me think that it somehow grants permission for me as a BAU user logging in to be able to bootstrap some extra application data like Releases. Maybe we could we list what gets bootstrapped as a result of this? It seems like it's going to be the Content database Users and Invites run on Startup. Also "...will automatically be able to be users to access the system as..." doesn't sound right.